homeassistant/components/hyperion/__init__.py

```diff
@@ -52,15 +51,14 @@
 # The get_hyperion_unique_id method will create a per-entity unique id when given the
 # server id, an instance number and a name.
 
-# hass.data format
-# ================
-#
-# hass.data[DOMAIN] = {
-#     <config_entry.entry_id>: {
-#         "ROOT_CLIENT": <Hyperion Client>,
-#         "ON_UNLOAD": [<callable>, ...],
-#     }
-# }
+type HyperionConfigEntry = ConfigEntry[HyperionData]
+
+
+class HyperionData(TypedDict):
```

Comment:
Why a typed dict over a dataclass? (apart from the fact that the code needs less changes?)